### PR TITLE
fix(QRCode): Fix the issue where the background color and font color …

### PR DESCRIPTION
**Pull request by biubiukam**
…of the QR code component do not synchronize with the theme in dark mode

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/47126

### 💡 Background and solution

Refer to other components to handle changes in style based on tokens, and use tokens at the QR code component layer to calculate transparency.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix mask background color and font color in dark mode      |
| 🇨🇳 Chinese |    修复深色模式下的遮罩背景颜色和字体颜色       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by kampiu** [comment]
Preview under default theme

image

Preview under dark themes

image

Can the document site add a theme switching function, or have I not noticed the existence of related functions?

File: components/qr-code/index.tsx [hunk 1]
```diff
@@ -78,25 +79,35 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
...
+  const fontColor = new TinyColor(token.colorText).toRgbString();
+
   return wrapCSSVar(
     <div
       className={mergedCls}
       style={{ ...style, width: size, height: size, backgroundColor: bgColor }}
     >
       {status !== 'active' && (
-        <div className={`${prefixCls}-mask`}>
+        <div className={`${prefixCls}-mask`} style={{ backgroundColor: maskBackgroundColor }}>
```
Comment (MadCcc) [inline components/qr-code/index.tsx:92]: 不要用内联 style，写到样式里

Comment (kampiu) [reply components/qr-code/index.tsx]: Okay, I have removed the inline style, adjusted it to style handling, and regenerated the test snapshot

File: components/qr-code/style/index.ts [hunk 2]
```diff
@@ -63,15 +64,16 @@ const genQRCodeStyle: GenerateStyle<QRCodeToken> = (token) => {
   };
 };
 
-export const prepareComponentToken: GetDefaultToken<'QRCode'> = () => ({});
+export const prepareComponentToken: GetDefaultToken<'QRCode'> = (token) => ({
+  QRCodeTextColor: new TinyColor(token.colorText).toRgbString(),
```
Comment (MadCcc) [inline components/qr-code/style/index.ts:68]: 这个不需要用 tinycolor 转换，还是写在下面
```suggestion
  QRCodeTextColor: token.colorText,
```

**Review by afc163 — APPROVED** [review]

**Pull request merged**